appwindow.js

```diff
@@ -257,8 +284,7 @@ var AppWindow = GObject.registerClass(
         }
 
         update_app_paintable() {
-            this.app_paintable = this.settings.get_boolean('transparent-background') &&
-                                 this.settings.get_double('background-opacity') < 1.0;
```

Comment:
Don't remove this line, it's correct. There is no point in doing custom painting if `background-opacity == 1.0`. If the window doesn't immediately become transparent after changing from `1.0`, try calling [`this.queue_draw()`](https://gjs-docs.gnome.org/gtk20~2.24.33/gtk.widget#method-queue_draw)

Reply:
Ok, then I will file the bug I mentioned above in a separate issue.

Reply:
Actually there is an outstanding issue with this. I tried to record this but actually the recording shows the proper function!?

Can you duplicate this:
1. Set transparency to 1.0
2. Hide/close window
3. Open window
4. Change transparency in settings
5. Type - you should see a weird ghosting effect on the letters, or some fully white blocks
6. Window does not refreshed until size is changed

Reply:
Please report it as a separate issue.

Have you tried adding `this.queue_draw()` to `update_app_paintable()`?

Reply:
I have tried queue_draw() to no success, we can continue the conversations here #73.